src/collapse/dtrace.rs

```diff
@@ -164,6 +169,21 @@ impl Folder {
         )
     }
 
+    fn demangle<'a>(&self, frame: &'a str) -> Cow<'a, str> {
+        let mut parts = frame.splitn(2, '`');
+        if let (Some(pname), Some(func)) = (parts.next(), parts.next()) {
+            if self.opt.includeoffset {
+                let mut parts = func.rsplitn(2, '+');
+                if let (Some(offset), Some(func)) = (parts.next(), parts.next()) {
+                    return Cow::Owned(format!("{}`{}+{}", pname, demangle(func), offset));
```

Comment:
This is a little confusing as this function is itself called `demangle`, so this looks like recursion, even though it is not. I think I'd prefer using `symbolic_demangle::demangle` explicitly.

Reply:
Yes, I can see how that looks like recursion.